gateway/gateway.nginx.conf

```diff
@@ -0,0 +1,71 @@
+map $http_upgrade $connection_upgrade {
+    default upgrade;
+    ''      close;
+}
+
+server {
+    listen 80 default_server;
+    listen [::]:80 default_server;
+    server_name _;
+
+    location /.well-known/acme-challenge {
+        proxy_pass http://letsencrypt;
+
+        proxy_set_header Host $host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Forwarded-Proto https;
+    }
+
+    location / {
+        return 301 https://$host$request_uri;
+    }
+}
+
+server {
+    server_name *.internal.hail.is;
+
+    location = /auth {
+        internal;
+        proxy_pass http://router-resolver/;
+        proxy_set_header Host $host;
+    }
+
+    location / {
+        auth_request /auth;
+        auth_request_set $router_ip $upstream_http_x_router_ip;
+        proxy_pass http://$router_ip$request_uri;
+
+        proxy_set_header Host $host;
+        proxy_http_version 1.1;
```

Comment:
We need to set 

```
proxy_set_header X-Forwarded-Proto $scheme;
```

Reply:
Else we cannot read the correct protocol from within the proxied service's server. This breaks notebook2, which relies on a callback url issued to auth0.